src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -151,6 +287,84 @@ private static string GetResourceString(ref Utf8JsonReader json, ExceptionResour
 
             return message;
         }
+
+        public static void ThrowJsonWriterException(ExceptionResource resource, int currentDepth = default, byte token = default, JsonTokenType tokenType = default)
+        {
+            throw GetJsonReaderException(resource, currentDepth, token, tokenType);
+        }
+
+        public static void ThrowJsonWriterException_InvalidUTF8(ReadOnlySpan<byte> value)
+        {
+            var builder = new StringBuilder();
+
+            int printFirst10 = Math.Min(value.Length, 10);
+
+            for (int i = 0; i < printFirst10; i++)
+            {
+                byte nextByte = value[i];
+                if (IsPrintable(nextByte))
+                {
+                    builder.Append((char)nextByte);
+                }
+                else
+                {
+                    builder.Append($"0x{nextByte:X2}");
+                }
+            }
```

Comment:
Should there be something like:
```C#
if (printFirst10 < value.Length)
{
    builder.Append("...");
}
```
to highlight that there's additional text not included?